modules/drivers/redshift/src/metabase/driver/redshift.clj

```diff
@@ -116,7 +115,9 @@
 
 (defmethod sql.qp/add-interval-honeysql-form :redshift
   [_ hsql-form amount unit]
-  (hsql/call :dateadd (hx/literal unit) amount (hx/->timestamp hsql-form)))
+  (let [hsql-form (hx/->timestamp hsql-form)]
+    (-> (hsql/call :dateadd (hx/literal unit) amount hsql-form)
+        (hx/with-type-info (hx/type-info hsql-form)))))
```

Comment:
Is this always returning the correct type info? What if hsql-form is a `date` type and the unit is `:second`. Then the return type will actually be `timestamp` but the type-info will still be `date`.

Reply:
we take the type after we cast it to timestamp so it should be fine.

Reply:
Ah, you're right. I didn't realise you updated the `hsql-form` argument. Never mind then.